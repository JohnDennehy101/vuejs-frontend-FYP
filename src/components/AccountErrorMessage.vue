<template>
  <div class="form-control" v-show="toggle" id="error-sign-up-notice">
    <i class="fas fa-exclamation-triangle"></i>
    <p>{{ errorMessage }}</p>

    <span @click="hideErrorMessage">
      <i class="fas fa-times-circle"></i>
    </span>
  </div>
</template>

<script>
export default {
  props: {
    toggle: {
      type: Boolean,
    },
    errorMessage: {
      type: String,
    },
  },
  data() {
    return {
      display: this.toggle,
    };
  },

  methods: {
    hideErrorMessage() {
      this.$emit("hideErrorMessage", true);
    },
  },
};
</script>

<style scoped lang="scss">
#error-sign-up-notice {
  border-radius: 5px;
  background-color: #e57373;
  color: white;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.5rem;

  p {
    margin-left: 0.5rem;
    margin-right: 0.5rem;
  }
  i {
    pointer-events: all;
  }
  .form-control {
    width: 80%;
    margin: 0.5rem auto;
  }
}
</style>
