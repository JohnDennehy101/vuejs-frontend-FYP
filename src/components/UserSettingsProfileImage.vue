<template>
  <div id="image-display-parent-container">
    <img data-testid="userImage" v-if="profileImage" :src="profileImage" />
    <img
      data-testid="defaultImage"
      v-else
      src="../assets/defaultUserImage.png"
    />

    <input
      type="file"
      id="image-upload"
      @change="previewFiles($event)"
      accept="image/*"
    />
    <label for="image-upload">Update Image</label>
  </div>
</template>

<script>
import userService from "../services/UserService";
export default {
  props: {
    userService: {
      default: userService,
    },
    userId: String,
    image: String,
  },
  data() {
    return {
      profileImage: this.image,
      id: this.userId,
      newImage: null,
      invalidImageUpload: null,
    };
  },
  methods: {
    async previewFiles(event) {
      this.newImage = event.target.files[0];
      this.$emit("updateUserImage", true);

      await this.updateUserImage();
    },

    async updateUserImage() {
      const response = await this.userService.uploadUserProfileImage(
        this.id,
        this.newImage
      );

      if ("error" in response) {
        this.invalidImageUpload = true;
      } else {
        this.$emit("updateUserImage", false);
        setTimeout(() => this.$router.go(0), 5000);
      }
    },
  },
};
</script>

<style scoped lang="scss">
#image-display-parent-container {
  width: 60%;
  margin: 1rem auto;
  text-align: center;
  border: 1px solid #eeeeee;

  @include for-phone-only {
    width: 90%;
  }
}
img {
  max-width: 250px;
  max-height: 250px;
  border-radius: 50%;
  display: block;
  margin: 2rem auto 2rem auto;
}
input[type="file"] {
  opacity: 0;
  z-index: -1;
  position: absolute;
}
label {
  display: flex;
  text-align: center;
  justify-content: center;
  align-items: center;
  border-radius: 10px;
  border: none;
  width: 30%;
  font-size: 1rem;
  height: 2.5rem;
  margin: 2rem auto;
  background-color: $primary-button-background-colour;
  color: $primary-button-text-colour;

  @include for-phone-only {
    width: 80%;
  }
}
label:hover {
  cursor: pointer;
}
</style>
