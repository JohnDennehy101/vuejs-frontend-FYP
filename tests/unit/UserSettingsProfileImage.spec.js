import { mount, shallowMount } from "@vue/test-utils";
import UserSettingsProfileImage from "@/components/UserSettingsProfileImage.vue";
import AccountErrorMessage from "@/components/AccountErrorMessage";
import mockSuccessfulUserService from "./mocks/userService.mock";
import mockStore from "./mocks/mockStore.mock";
import mockRouter from "./mocks/mockRouter.mock";
import { nextTick } from "vue";

const mockRoute = {
  params: {
    id: 1,
  },
};

const mockFailureUserService = {
  uploadUserProfileImage() {
    return new Promise((resolve, reject) => {
      resolve({
        error: {
          message: "Error adding",
        },
      });
    });
  },
};

describe("UserSettingsProfileImage.vue", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should render correctly", () => {
    const wrapper = shallowMount(UserSettingsProfileImage, {
      props: {
        userService: mockSuccessfulUserService,
        id: "1",
        image: "https://www.cloudinaryimage.com",
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("user image rendered if passed as prop in component", async () => {
    const wrapper = mount(UserSettingsProfileImage, {
      props: {
        userService: mockSuccessfulUserService,
        id: "1",
        image: "https://www.cloudinary.com",
      },
    });

    const profileImage = await wrapper.find('[data-testid="userImage"]');

    console.log(profileImage);

    await nextTick();

    expect(profileImage.isVisible()).toBe(true);
    expect(wrapper.find("defaultImage").exists()).toBe(false);
  });

  it("default image rendered if user image not passed as prop in component", async () => {
    const wrapper = mount(UserSettingsProfileImage, {
      props: {
        userService: mockSuccessfulUserService,
        id: "1",
        image: null,
      },
    });

    const profileImage = await wrapper.find('[data-testid="userImage"]');

    console.log(profileImage);

    await nextTick();

    expect(profileImage.exists()).toBe(false);
    expect(wrapper.find('[data-testid="defaultImage"]').exists()).toBe(true);
  });

  it("successfully updates user if file provided", async () => {
    const wrapper = mount(UserSettingsProfileImage, {
      props: {
        userService: mockSuccessfulUserService,
        userId: "1",
        image: null,
      },
      global: {
        mocks: {
          $store: mockStore,
          $route: mockRoute,
          $router: mockRouter,
        },
      },
    });

    let imageInputElement = await wrapper.find("#image-upload");
    imageInputElement.trigger("change");

    expect(wrapper.emitted()).toHaveProperty("updateUserImage");
  });

  it("does not updates user if error received in response", async () => {
    const wrapper = mount(UserSettingsProfileImage, {
      props: {
        userService: mockFailureUserService,
        userId: "1",
        image: null,
      },
      global: {
        mocks: {
          $store: mockStore,
          $route: mockRoute,
          $router: mockRouter,
        },
      },
    });

    let imageInputElement = await wrapper.find("#image-upload");
    await imageInputElement.trigger("change");

    expect(wrapper.vm.invalidImageUpload).toBe(true);

    expect(wrapper.emitted()).toHaveProperty("updateUserImage");
  });
});
