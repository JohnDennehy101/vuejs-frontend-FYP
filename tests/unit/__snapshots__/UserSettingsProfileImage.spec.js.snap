// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSettingsProfileImage.vue should render correctly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        id="1"
      >
        <img
          data-testid="userImage"
          src="https://www.cloudinaryimage.com"
        />
        <input
          accept="image/*"
          id="image-upload"
          type="file"
        />
        <label
          for="image-upload"
        >
          Update Image
        </label>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {
          "id": undefined,
          "image": "https://www.cloudinaryimage.com",
          "invalidImageUpload": null,
          "jwt": undefined,
          "newImage": null,
          "previewFiles": [Function],
          "profileImage": "https://www.cloudinaryimage.com",
          "token": undefined,
          "updateUserImage": [Function],
          "userId": undefined,
          "userService": Object {
            "loginUser": [Function],
            "registerUser": [Function],
            "updateUser": [Function],
            "uploadUserProfileImage": [Function],
          },
        },
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {
            "id": undefined,
            "image": "https://www.cloudinaryimage.com",
            "invalidImageUpload": null,
            "jwt": undefined,
            "newImage": null,
            "previewFiles": [Function],
            "profileImage": "https://www.cloudinaryimage.com",
            "token": undefined,
            "updateUserImage": [Function],
            "userId": undefined,
            "userService": Object {
              "loginUser": [Function],
              "registerUser": [Function],
              "updateUser": [Function],
              "uploadUserProfileImage": [Function],
            },
          },
          "_value": Object {
            "id": undefined,
            "image": "https://www.cloudinaryimage.com",
            "invalidImageUpload": null,
            "jwt": undefined,
            "newImage": null,
            "previewFiles": [Function],
            "profileImage": "https://www.cloudinaryimage.com",
            "token": undefined,
            "updateUserImage": [Function],
            "userId": undefined,
            "userService": Object {
              "loginUser": [Function],
              "registerUser": [Function],
              "updateUser": [Function],
              "uploadUserProfileImage": [Function],
            },
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {
          "id": undefined,
          "image": "https://www.cloudinaryimage.com",
          "invalidImageUpload": null,
          "jwt": undefined,
          "newImage": null,
          "previewFiles": [Function],
          "profileImage": "https://www.cloudinaryimage.com",
          "token": undefined,
          "updateUserImage": [Function],
          "userId": undefined,
          "userService": Object {
            "loginUser": [Function],
            "registerUser": [Function],
            "updateUser": [Function],
            "uploadUserProfileImage": [Function],
          },
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "id": undefined,
          "image": "https://www.cloudinaryimage.com",
          "invalidImageUpload": null,
          "jwt": undefined,
          "newImage": null,
          "previewFiles": [Function],
          "profileImage": "https://www.cloudinaryimage.com",
          "token": undefined,
          "updateUserImage": [Function],
          "userId": undefined,
          "userService": Object {
            "loginUser": [Function],
            "registerUser": [Function],
            "updateUser": [Function],
            "uploadUserProfileImage": [Function],
          },
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {
          "id": undefined,
          "image": "https://www.cloudinaryimage.com",
          "invalidImageUpload": null,
          "jwt": undefined,
          "newImage": null,
          "previewFiles": [Function],
          "profileImage": "https://www.cloudinaryimage.com",
          "token": undefined,
          "updateUserImage": [Function],
          "userId": undefined,
          "userService": Object {
            "loginUser": [Function],
            "registerUser": [Function],
            "updateUser": [Function],
            "uploadUserProfileImage": [Function],
          },
        },
      },
      "slots": Object {},
      "slotsProxy": null,
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "__v_isReadonly": 0,
            "__v_isRef": 0,
            "__v_isShallow": 0,
            "__v_raw": 0,
            "__v_skip": 4,
            "_isMockFunction": 0,
            "constructor": 0,
            "id": 2,
            "image": 3,
            "invalidImageUpload": 2,
            "isVueInstance": 0,
            "jwt": 3,
            "newImage": 2,
            "previewFiles": 4,
            "profileImage": 2,
            "toJSON": 0,
            "token": 2,
            "updateUserImage": 4,
            "userId": 3,
            "userService": 3,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {
            "id": "1",
          },
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "id": undefined,
            "image": "https://www.cloudinaryimage.com",
            "invalidImageUpload": null,
            "jwt": undefined,
            "newImage": null,
            "previewFiles": [Function],
            "profileImage": "https://www.cloudinaryimage.com",
            "token": undefined,
            "updateUserImage": [Function],
            "userId": undefined,
            "userService": Object {
              "loginUser": [Function],
              "registerUser": [Function],
              "updateUser": [Function],
              "uploadUserProfileImage": [Function],
            },
          },
          "da": null,
          "data": Object {
            "id": undefined,
            "invalidImageUpload": null,
            "newImage": null,
            "profileImage": "https://www.cloudinaryimage.com",
            "token": undefined,
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "image": "https://www.cloudinaryimage.com",
            "jwt": undefined,
            "userId": undefined,
            "userService": Object {
              "loginUser": [Function],
              "registerUser": [Function],
              "updateUser": [Function],
              "uploadUserProfileImage": [Function],
            },
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "image": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "jwt": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "userId": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "userService": Object {
                "0": false,
                "1": true,
                "default": UserService {
                  "baseUrl": undefined,
                },
              },
            },
            Array [
              "userService",
            ],
          ],
          "provides": Object {},
          "proxy": Object {
            "id": undefined,
            "image": "https://www.cloudinaryimage.com",
            "invalidImageUpload": null,
            "jwt": undefined,
            "newImage": null,
            "previewFiles": [Function],
            "profileImage": "https://www.cloudinaryimage.com",
            "token": undefined,
            "updateUserImage": [Function],
            "userId": undefined,
            "userService": Object {
              "loginUser": [Function],
              "registerUser": [Function],
              "updateUser": [Function],
              "uploadUserProfileImage": [Function],
            },
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "slotsProxy": null,
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "ce": undefined,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": Array [],
                "dynamicProps": Array [
                  "src",
                ],
                "el": <img
                  data-testid="userImage"
                  src="https://www.cloudinaryimage.com"
                />,
                "key": 0,
                "patchFlag": 8,
                "props": Object {
                  "data-testid": "userImage",
                  "key": 0,
                  "src": "https://www.cloudinaryimage.com",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "img",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  accept="image/*"
                  id="image-upload"
                  type="file"
                />,
                "key": null,
                "patchFlag": 32,
                "props": Object {
                  "accept": "image/*",
                  "id": "image-upload",
                  "onChange": [Function],
                  "type": "file",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "ce": undefined,
                "children": "Update Image",
                "component": null,
                "ctx": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <label
                  for="image-upload"
                >
                  Update Image
                </label>,
                "key": null,
                "patchFlag": -1,
                "props": Object {
                  "for": "image-upload",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": Array [],
                "dynamicProps": Array [
                  "src",
                ],
                "el": <img
                  data-testid="userImage"
                  src="https://www.cloudinaryimage.com"
                />,
                "key": 0,
                "patchFlag": 8,
                "props": Object {
                  "data-testid": "userImage",
                  "key": 0,
                  "src": "https://www.cloudinaryimage.com",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "img",
              },
            ],
            "dynamicProps": null,
            "el": <div
              id="1"
            >
              <img
                data-testid="userImage"
                src="https://www.cloudinaryimage.com"
              />
              <input
                accept="image/*"
                id="image-upload"
                type="file"
              />
              <label
                for="image-upload"
              >
                Update Image
              </label>
            </div>,
            "key": null,
            "patchFlag": 16,
            "props": Object {
              "id": "1",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "data": [Function],
            "methods": Object {
              "previewFiles": [Function],
              "updateUserImage": [Function],
            },
            "props": Object {
              "image": [Function],
              "jwt": [Function],
              "userId": [Function],
              "userService": Object {
                "default": UserService {
                  "baseUrl": undefined,
                },
              },
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          id="1"
        >
          <img
            data-testid="userImage"
            src="https://www.cloudinaryimage.com"
          />
          <input
            accept="image/*"
            id="image-upload"
            type="file"
          />
          <label
            for="image-upload"
          >
            Update Image
          </label>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "id": "1",
          "image": "https://www.cloudinaryimage.com",
          "ref": "VTU_COMPONENT",
          "userService": Object {
            "loginUser": [Function],
            "registerUser": [Function],
            "updateUser": [Function],
            "uploadUserProfileImage": [Function],
          },
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "data": [Function],
          "methods": Object {
            "previewFiles": [Function],
            "updateUserImage": [Function],
          },
          "props": Object {
            "image": [Function],
            "jwt": [Function],
            "userId": [Function],
            "userService": Object {
              "default": UserService {
                "baseUrl": undefined,
              },
            },
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          id="1"
        >
          <img
            data-testid="userImage"
            src="https://www.cloudinaryimage.com"
          />
          <input
            accept="image/*"
            id="image-upload"
            type="file"
          />
          <label
            for="image-upload"
          >
            Update Image
          </label>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": Object {
    "id": undefined,
    "image": "https://www.cloudinaryimage.com",
    "invalidImageUpload": null,
    "jwt": undefined,
    "newImage": null,
    "previewFiles": [Function],
    "profileImage": "https://www.cloudinaryimage.com",
    "token": undefined,
    "updateUserImage": [Function],
    "userId": undefined,
    "userService": Object {
      "loginUser": [Function],
      "registerUser": [Function],
      "updateUser": [Function],
      "uploadUserProfileImage": [Function],
    },
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "id": undefined,
      "image": "https://www.cloudinaryimage.com",
      "invalidImageUpload": null,
      "jwt": undefined,
      "newImage": null,
      "previewFiles": [Function],
      "profileImage": "https://www.cloudinaryimage.com",
      "token": undefined,
      "updateUserImage": [Function],
      "userId": undefined,
      "userService": Object {
        "loginUser": [Function],
        "registerUser": [Function],
        "updateUser": [Function],
        "uploadUserProfileImage": [Function],
      },
    },
  },
  "wrapperElement": <div
    id="1"
  >
    <img
      data-testid="userImage"
      src="https://www.cloudinaryimage.com"
    />
    <input
      accept="image/*"
      id="image-upload"
      type="file"
    />
    <label
      for="image-upload"
    >
      Update Image
    </label>
  </div>,
}
`;
